src/hapi/pipelines/database/humanitarian_needs.py

```diff
@@ -63,44 +85,37 @@ def populate(self) -> None:
         self._metadata.add_dataset(dataset)
         dataset_id = dataset["id"]
         dataset_name = dataset["name"]
-        resource = dataset.get_resource(
-            1
-        )  # assumes second resource is latest!
+        resource = dataset.get_resource(0)  # assumes first resource is latest!
         self._metadata.add_resource(dataset_id, resource)
         negative_values_by_iso3 = {}
         rounded_values_by_iso3 = {}
         resource_id = resource["id"]
         resource_name = resource["name"]
-        year = int(resource_name[-15:-11])
+        year = int(resource_name[-4:])
         time_period_start = datetime(year, 1, 1)
         time_period_end = datetime(year, 12, 31, 23, 59, 59)
         url = resource["url"]
         headers, rows = reader.get_tabular_rows(url, dict_form=True)
         # Admin 1 PCode,Admin 2 PCode,Sector,Gender,Age Group,Disabled,Population Group,Population,In Need,Targeted,Affected,Reached
         for row in rows:
-            admin2_ref = self.get_admin2_ref(row, dataset_name, errors)
-            if not admin2_ref:
-                continue
             countryiso3 = row["Country ISO3"]
-            population_group = row["Population Group"]
-            if population_group == "ALL":
-                population_group = "all"
+            admin2_ref = self.get_admin2_ref(row, dataset_name, errors)
+            provider_admin1_name = row["Admin 1 Name"]
+            if provider_admin1_name is None:
+                provider_admin1_name = ""
+            provider_admin2_name = row["Admin 2 Name"]
+            if provider_admin2_name is None:
+                provider_admin2_name = ""
```

Comment:
You can use the [get_provider_name](https://github.com/OCHA-DAP/hapi-pipelines/blob/a37ffa1a791e6957f36d625d8a9080e2f9389298/src/hapi/pipelines/utilities/provider_admin_names.py#L4) function for this

Reply:
Good idea!